discovery-infra/logger.py

```diff
@@ -17,6 +18,16 @@ def format(self, record):
         return self._filter(original)
 
 
+class suppressLog(suppress):
```

Comment:
it's a misleading name, I thought it just suppresses logging inside the block
Maybe ``suppressAndLog`` or ``suppressButLog``? Or ``suppressWithNotification``/``WithLogging``